middleware.js

```diff
@@ -1,4 +1,45 @@
 var http = require('http');
+var url = require('url');
 
-exports.createServer = function () {
+var app = {};
+
+app.use = function (route, fn) {
+    if ('function' == typeof route) {
+        fn = route;
+        route = '/';
+    }
+
+    this.handlers.push({route: route, handleFn: fn});
 };
+
+app.handle = function (req, res) {
+    var index = 0;
+    var path = url.parse(req.url).pathname;
+    if (undefined == path) path = '/';
+
+    function next() {
+        var handler = app.handlers[index++];
```

Comment:
Notes the following 2 points:

- the `++` is not clear to represent what the statement wants
- we should make sure `index` gets changed iff(if and only if) one `next` function is going to quit.